pyproject.toml

```diff
@@ -45,7 +45,7 @@ dependencies = [
 	"semver==3.0.0-dev.4"
 ]
 [project.optional-dependencies]
-dev = ["pep8-naming==0.7.0", "PyGithub==1.55", "flake8==3.8.1", "pyflakes==2.2.0", "pytest>=3.6", "pre-commit==2.20.0"]
+dev = ["pep8-naming==0.7.0", "PyGithub==1.55", "flake8==3.9.1", "pyflakes==2.3.0", "pytest>=3.6", "pre-commit==2.20.0"]
```

Comment:
`make` needed adjustments for py3.9. Not sure why this wasn't caught in other similar bug reports.

Reply:
I have no issue with upgrading the package versions, but for what it is worth, on python 3.9.18 on Ubuntu 22.04 I did not have to update the requirements. :shrug: 

Make appeared to build the package fine even with:
```shell
Collecting flake8==3.8.1
  Downloading flake8-3.8.1-py2.py3-none-any.whl.metadata (3.7 kB)
```

<details><summary>Shell Output</summary>
<p>

```shell
detection-rules on  package_kql_kibana [!] is  v0.1.0 via  v3.8.18 (eql-rules-build) on  eric.forte 
❯ make
pip3 install virtualenv
Defaulting to user installation because normal site-packages is not writeable
Looking in indexes: https://pypi.org/simple, https://eric.forte%40elastic.co:****@artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Requirement already satisfied: virtualenv in /home/forteea1/.local/lib/python3.8/site-packages (20.24.7)
Requirement already satisfied: distlib<1,>=0.3.7 in /home/forteea1/.local/lib/python3.8/site-packages (from virtualenv) (0.3.7)
Requirement already satisfied: filelock<4,>=3.12.2 in /home/forteea1/.local/lib/python3.8/site-packages (from virtualenv) (3.13.1)
Requirement already satisfied: platformdirs<5,>=3.9.1 in /home/forteea1/.local/lib/python3.8/site-packages (from virtualenv) (3.11.0)
virtualenv ./env/detection-rules-build --python=python3.9
created virtual environment CPython3.9.18.final.0-64 in 1677ms
  creator CPython3Posix(dest=/home/forteea1/Code/clean_mains/detection-rules/env/detection-rules-build, clear=False, no_vcs_ignore=False, global=False)
  seeder FromAppData(download=False, pip=bundle, setuptools=bundle, wheel=bundle, via=copy, app_data_dir=/home/forteea1/.local/share/virtualenv)
    added seed packages: pip==23.3.1, setuptools==68.2.2, wheel==0.41.3
  activators BashActivator,CShellActivator,FishActivator,NushellActivator,PowerShellActivator,PythonActivator
./env/detection-rules-build/bin/python -m pip install setuptools -U
Looking in indexes: https://pypi.org/simple, https://eric.forte%40elastic.co:****@artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Requirement already satisfied: setuptools in ./env/detection-rules-build/lib/python3.9/site-packages (68.2.2)
Collecting setuptools
  Using cached setuptools-69.1.1-py3-none-any.whl.metadata (6.2 kB)
Using cached setuptools-69.1.1-py3-none-any.whl (819 kB)
Installing collected packages: setuptools
  Attempting uninstall: setuptools
    Found existing installation: setuptools 68.2.2
    Uninstalling setuptools-68.2.2:
      Successfully uninstalled setuptools-68.2.2
Successfully installed setuptools-69.1.1

[notice] A new release of pip is available: 23.3.1 -> 24.0
[notice] To update, run: /home/forteea1/Code/clean_mains/detection-rules/env/detection-rules-build/bin/python -m pip install --upgrade pip
Installing kql and kibana packages...
./env/detection-rules-build/bin/python -m pip install -e tools/kql tools/kibana
Looking in indexes: https://pypi.org/simple, https://eric.forte%40elastic.co:****@artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Processing ./tools/kibana
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Obtaining file:///home/forteea1/Code/clean_mains/detection-rules/tools/kql
  Installing build dependencies ... done
  Checking if build backend supports build_editable ... done
  Getting requirements to build editable ... done
  Preparing editable metadata (pyproject.toml) ... done
Collecting requests<3.0,>=2.25 (from detection-rules-kibana==0.1.0)
  Using cached requests-2.31.0-py3-none-any.whl.metadata (4.6 kB)
Collecting elasticsearch~=8.1 (from detection-rules-kibana==0.1.0)
  Downloading elasticsearch-8.12.1-py3-none-any.whl.metadata (5.3 kB)
Collecting eql==0.9.19 (from detection-rules-kql==0.1.6)
  Using cached eql-0.9.19-py2.py3-none-any.whl.metadata (3.7 kB)
Collecting lark-parser>=0.11.1 (from detection-rules-kql==0.1.6)
  Using cached https://artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/lark-parser/0.12.0/lark_parser-0.12.0-py2.py3-none-any.whl (103 kB)
Collecting elastic-transport<9,>=8 (from elasticsearch~=8.1->detection-rules-kibana==0.1.0)
  Downloading elastic_transport-8.12.0-py3-none-any.whl.metadata (3.5 kB)
Collecting charset-normalizer<4,>=2 (from requests<3.0,>=2.25->detection-rules-kibana==0.1.0)
  Downloading charset_normalizer-3.3.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (33 kB)
Collecting idna<4,>=2.5 (from requests<3.0,>=2.25->detection-rules-kibana==0.1.0)
  Using cached idna-3.6-py3-none-any.whl.metadata (9.9 kB)
Collecting urllib3<3,>=1.21.1 (from requests<3.0,>=2.25->detection-rules-kibana==0.1.0)
  Using cached urllib3-2.2.1-py3-none-any.whl.metadata (6.4 kB)
Collecting certifi>=2017.4.17 (from requests<3.0,>=2.25->detection-rules-kibana==0.1.0)
  Using cached certifi-2024.2.2-py3-none-any.whl.metadata (2.2 kB)
Using cached eql-0.9.19-py2.py3-none-any.whl (105 kB)
Downloading elasticsearch-8.12.1-py3-none-any.whl (432 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 432.1/432.1 kB 10.0 MB/s eta 0:00:00
Using cached requests-2.31.0-py3-none-any.whl (62 kB)
Using cached certifi-2024.2.2-py3-none-any.whl (163 kB)
Downloading charset_normalizer-3.3.2-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (142 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 142.3/142.3 kB 14.1 MB/s eta 0:00:00
Downloading elastic_transport-8.12.0-py3-none-any.whl (59 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.9/59.9 kB 12.4 MB/s eta 0:00:00
Using cached idna-3.6-py3-none-any.whl (61 kB)
Using cached urllib3-2.2.1-py3-none-any.whl (121 kB)
Building wheels for collected packages: detection-rules-kibana, detection-rules-kql
  Building wheel for detection-rules-kibana (pyproject.toml) ... done
  Created wheel for detection-rules-kibana: filename=detection_rules_kibana-0.1.0-py3-none-any.whl size=6841 sha256=a31b93273370ea9f153bed273d7b8b2c79dc913368e968bd4b4dd2790e395378
  Stored in directory: /tmp/pip-ephem-wheel-cache-9iunbplf/wheels/87/dc/ba/2db4b1c93f38b3de4a2f732edc00b72967058bba39fce751c7
  Building editable for detection-rules-kql (pyproject.toml) ... done
  Created wheel for detection-rules-kql: filename=detection_rules_kql-0.1.6-0.editable-py3-none-any.whl size=2970 sha256=c7e014040cd310a11d08d2ebcfd256db1eeed9d628f8502b31420cbcd59abd10
  Stored in directory: /tmp/pip-ephem-wheel-cache-9iunbplf/wheels/be/aa/c7/aea8d548aa666e33dd54af8162142d0c65ec95755b4874579f
Successfully built detection-rules-kibana detection-rules-kql
Installing collected packages: lark-parser, urllib3, idna, eql, charset-normalizer, certifi, requests, elastic-transport, detection-rules-kql, elasticsearch, detection-rules-kibana
Successfully installed certifi-2024.2.2 charset-normalizer-3.3.2 detection-rules-kibana-0.1.0 detection-rules-kql-0.1.6 elastic-transport-8.12.0 elasticsearch-8.12.1 eql-0.9.19 idna-3.6 lark-parser-0.12.0 requests-2.31.0 urllib3-2.2.1

[notice] A new release of pip is available: 23.3.1 -> 24.0
[notice] To update, run: /home/forteea1/Code/clean_mains/detection-rules/env/detection-rules-build/bin/python -m pip install --upgrade pip
Installing all dependencies...
./env/detection-rules-build/bin/python -m pip install .[dev]
Looking in indexes: https://pypi.org/simple, https://eric.forte%40elastic.co:****@artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Processing /home/forteea1/Code/clean_mains/detection-rules
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting Click~=8.1.0
  Using cached click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
Requirement already satisfied: elasticsearch~=8.1 in ./env/detection-rules-build/lib/python3.9/site-packages (8.12.1)
Requirement already satisfied: eql==0.9.19 in ./env/detection-rules-build/lib/python3.9/site-packages (0.9.19)
Collecting jsl==0.2.4
  Using cached jsl-0.2.4-py3-none-any.whl
Collecting jsonschema>=3.2.0
  Using cached jsonschema-4.21.1-py3-none-any.whl.metadata (7.8 kB)
Collecting marko==2.0.1
  Using cached marko-2.0.1-py3-none-any.whl.metadata (4.5 kB)
Collecting marshmallow-dataclass~=8.5.12 (from marshmallow-dataclass[union]~=8.5.12)
  Downloading marshmallow_dataclass-8.5.14-py3-none-any.whl.metadata (12 kB)
Collecting marshmallow-jsonschema~=0.12.0
  Downloading marshmallow_jsonschema-0.12.0-py3-none-any.whl.metadata (7.5 kB)
Collecting marshmallow-union~=0.1.15
  Downloading marshmallow_union-0.1.15.post1-py2.py3-none-any.whl.metadata (2.9 kB)
Collecting marshmallow~=3.13.0
  Downloading marshmallow-3.13.0-py2.py3-none-any.whl.metadata (7.8 kB)
Collecting pytoml==0.1.21
  Downloading pytoml-0.1.21-py2.py3-none-any.whl.metadata (2.0 kB)
Requirement already satisfied: requests~=2.27 in ./env/detection-rules-build/lib/python3.9/site-packages (2.31.0)
Collecting toml==0.10.0
  Downloading toml-0.10.0-py2.py3-none-any.whl.metadata (5.5 kB)
Collecting typing-inspect==0.8.0
  Downloading typing_inspect-0.8.0-py3-none-any.whl.metadata (1.5 kB)
Collecting XlsxWriter~=1.3.6
  Downloading XlsxWriter-1.3.9-py2.py3-none-any.whl.metadata (2.5 kB)
Collecting semver==3.0.0-dev.4
  Downloading semver-3.0.0.dev4-py3-none-any.whl.metadata (5.7 kB)
Collecting typing-extensions==4.5.0
  Downloading typing_extensions-4.5.0-py3-none-any.whl.metadata (8.5 kB)
Collecting PyYAML~=6.0.1
  Downloading PyYAML-6.0.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (2.1 kB)
Collecting pep8-naming==0.7.0
  Downloading pep8_naming-0.7.0-py2.py3-none-any.whl.metadata (3.4 kB)
Collecting PyGithub==1.55
  Downloading PyGithub-1.55-py3-none-any.whl.metadata (1.8 kB)
Collecting flake8==3.8.1
  Downloading flake8-3.8.1-py2.py3-none-any.whl.metadata (3.7 kB)
Collecting pyflakes==2.2.0
  Downloading pyflakes-2.2.0-py2.py3-none-any.whl.metadata (3.9 kB)
Collecting pytest>=3.6
  Downloading pytest-8.0.2-py3-none-any.whl.metadata (7.7 kB)
Collecting pre-commit==2.20.0
  Downloading pre_commit-2.20.0-py2.py3-none-any.whl.metadata (1.8 kB)
Requirement already satisfied: lark-parser~=0.12.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from eql==0.9.19) (0.12.0)
Collecting pycodestyle<2.7.0,>=2.6.0a1 (from flake8==3.8.1)
  Downloading pycodestyle-2.6.0-py2.py3-none-any.whl.metadata (30 kB)
Collecting mccabe<0.7.0,>=0.6.0 (from flake8==3.8.1)
  Downloading mccabe-0.6.1-py2.py3-none-any.whl.metadata (4.3 kB)
Collecting flake8-polyfill<2,>=1.0.2 (from pep8-naming==0.7.0)
  Downloading flake8_polyfill-1.0.2-py2.py3-none-any.whl.metadata (4.0 kB)
Collecting cfgv>=2.0.0 (from pre-commit==2.20.0)
  Using cached cfgv-3.4.0-py2.py3-none-any.whl.metadata (8.5 kB)
Collecting identify>=1.0.0 (from pre-commit==2.20.0)
  Downloading identify-2.5.35-py2.py3-none-any.whl.metadata (4.4 kB)
Collecting nodeenv>=0.11.1 (from pre-commit==2.20.0)
  Using cached nodeenv-1.8.0-py2.py3-none-any.whl.metadata (21 kB)
Collecting virtualenv>=20.0.8 (from pre-commit==2.20.0)
  Downloading virtualenv-20.25.1-py3-none-any.whl.metadata (4.4 kB)
Collecting deprecated (from PyGithub==1.55)
  Using cached Deprecated-1.2.14-py2.py3-none-any.whl.metadata (5.4 kB)
Collecting pyjwt>=2.0 (from PyGithub==1.55)
  Using cached PyJWT-2.8.0-py3-none-any.whl.metadata (4.2 kB)
Collecting pynacl>=1.4.0 (from PyGithub==1.55)
  Downloading PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl.metadata (8.6 kB)
Collecting mypy-extensions>=0.3.0 (from typing-inspect==0.8.0)
  Downloading mypy_extensions-1.0.0-py3-none-any.whl.metadata (1.1 kB)
Requirement already satisfied: elastic-transport<9,>=8 in ./env/detection-rules-build/lib/python3.9/site-packages (from elasticsearch~=8.1) (8.12.0)
Collecting attrs>=22.2.0 (from jsonschema>=3.2.0)
  Using cached attrs-23.2.0-py3-none-any.whl.metadata (9.5 kB)
Collecting jsonschema-specifications>=2023.03.6 (from jsonschema>=3.2.0)
  Using cached jsonschema_specifications-2023.12.1-py3-none-any.whl.metadata (3.0 kB)
Collecting referencing>=0.28.4 (from jsonschema>=3.2.0)
  Using cached referencing-0.33.0-py3-none-any.whl.metadata (2.7 kB)
Collecting rpds-py>=0.7.1 (from jsonschema>=3.2.0)
  Downloading rpds_py-0.18.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (4.1 kB)
Collecting typeguard<4.0.0,>=2.4.1 (from marshmallow-dataclass[union]~=8.5.12)
  Downloading typeguard-3.0.2-py3-none-any.whl.metadata (3.7 kB)
Collecting iniconfig (from pytest>=3.6)
  Downloading iniconfig-2.0.0-py3-none-any.whl.metadata (2.6 kB)
Collecting packaging (from pytest>=3.6)
  Using cached packaging-23.2-py3-none-any.whl.metadata (3.2 kB)
Collecting pluggy<2.0,>=1.3.0 (from pytest>=3.6)
  Using cached pluggy-1.4.0-py3-none-any.whl.metadata (4.3 kB)
Collecting exceptiongroup>=1.0.0rc8 (from pytest>=3.6)
  Using cached exceptiongroup-1.2.0-py3-none-any.whl.metadata (6.6 kB)
Collecting tomli>=1.0.0 (from pytest>=3.6)
  Using cached tomli-2.0.1-py3-none-any.whl.metadata (8.9 kB)
Requirement already satisfied: charset-normalizer<4,>=2 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (3.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (2.2.1)
Requirement already satisfied: certifi>=2017.4.17 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (2024.2.2)
Requirement already satisfied: setuptools in ./env/detection-rules-build/lib/python3.9/site-packages (from nodeenv>=0.11.1->pre-commit==2.20.0) (69.1.1)
Collecting cffi>=1.4.1 (from pynacl>=1.4.0->PyGithub==1.55)
  Using cached cffi-1.16.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (1.5 kB)
Collecting importlib-metadata>=3.6 (from typeguard<4.0.0,>=2.4.1->marshmallow-dataclass[union]~=8.5.12)
  Downloading importlib_metadata-7.0.2-py3-none-any.whl.metadata (4.6 kB)
Collecting distlib<1,>=0.3.7 (from virtualenv>=20.0.8->pre-commit==2.20.0)
  Downloading distlib-0.3.8-py2.py3-none-any.whl.metadata (5.1 kB)
Collecting filelock<4,>=3.12.2 (from virtualenv>=20.0.8->pre-commit==2.20.0)
  Using cached filelock-3.13.1-py3-none-any.whl.metadata (2.8 kB)
Collecting platformdirs<5,>=3.9.1 (from virtualenv>=20.0.8->pre-commit==2.20.0)
  Downloading platformdirs-4.2.0-py3-none-any.whl.metadata (11 kB)
Collecting wrapt<2,>=1.10 (from deprecated->PyGithub==1.55)
  Downloading wrapt-1.16.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl.metadata (6.6 kB)
Collecting pycparser (from cffi>=1.4.1->pynacl>=1.4.0->PyGithub==1.55)
  Using cached https://artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/pycparser/2.21/pycparser-2.21-py2.py3-none-any.whl (118 kB)
Collecting zipp>=0.5 (from importlib-metadata>=3.6->typeguard<4.0.0,>=2.4.1->marshmallow-dataclass[union]~=8.5.12)
  Using cached zipp-3.17.0-py3-none-any.whl.metadata (3.7 kB)
Using cached flake8-3.8.1-py2.py3-none-any.whl (72 kB)
Using cached marko-2.0.1-py3-none-any.whl (41 kB)
Using cached pep8_naming-0.7.0-py2.py3-none-any.whl (7.7 kB)
Using cached pre_commit-2.20.0-py2.py3-none-any.whl (199 kB)
Using cached pyflakes-2.2.0-py2.py3-none-any.whl (66 kB)
Using cached PyGithub-1.55-py3-none-any.whl (291 kB)
Using cached pytoml-0.1.21-py2.py3-none-any.whl (8.5 kB)
Using cached semver-3.0.0.dev4-py3-none-any.whl (16 kB)
Using cached toml-0.10.0-py2.py3-none-any.whl (25 kB)
Using cached typing_extensions-4.5.0-py3-none-any.whl (27 kB)
Using cached typing_inspect-0.8.0-py3-none-any.whl (8.7 kB)
Using cached click-8.1.7-py3-none-any.whl (97 kB)
Using cached jsonschema-4.21.1-py3-none-any.whl (85 kB)
Using cached marshmallow-3.13.0-py2.py3-none-any.whl (47 kB)
Using cached marshmallow_dataclass-8.5.14-py3-none-any.whl (17 kB)
Using cached marshmallow_jsonschema-0.12.0-py3-none-any.whl (11 kB)
Using cached marshmallow_union-0.1.15.post1-py2.py3-none-any.whl (4.6 kB)
Downloading pytest-8.0.2-py3-none-any.whl (333 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 334.0/334.0 kB 9.9 MB/s eta 0:00:00
Downloading PyYAML-6.0.1-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (738 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 738.9/738.9 kB 14.8 MB/s eta 0:00:00
Using cached XlsxWriter-1.3.9-py2.py3-none-any.whl (145 kB)
Using cached attrs-23.2.0-py3-none-any.whl (60 kB)
Using cached cfgv-3.4.0-py2.py3-none-any.whl (7.2 kB)
Using cached exceptiongroup-1.2.0-py3-none-any.whl (16 kB)
Using cached flake8_polyfill-1.0.2-py2.py3-none-any.whl (7.3 kB)
Downloading identify-2.5.35-py2.py3-none-any.whl (98 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.9/98.9 kB 12.8 MB/s eta 0:00:00
Using cached jsonschema_specifications-2023.12.1-py3-none-any.whl (18 kB)
Using cached mccabe-0.6.1-py2.py3-none-any.whl (8.6 kB)
Using cached mypy_extensions-1.0.0-py3-none-any.whl (4.7 kB)
Using cached nodeenv-1.8.0-py2.py3-none-any.whl (22 kB)
Using cached pluggy-1.4.0-py3-none-any.whl (20 kB)
Using cached pycodestyle-2.6.0-py2.py3-none-any.whl (41 kB)
Using cached PyJWT-2.8.0-py3-none-any.whl (22 kB)
Using cached PyNaCl-1.5.0-cp36-abi3-manylinux_2_17_x86_64.manylinux2014_x86_64.manylinux_2_24_x86_64.whl (856 kB)
Using cached referencing-0.33.0-py3-none-any.whl (26 kB)
Downloading rpds_py-0.18.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (1.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.1/1.1 MB 14.4 MB/s eta 0:00:00
Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
Using cached typeguard-3.0.2-py3-none-any.whl (30 kB)
Downloading virtualenv-20.25.1-py3-none-any.whl (3.8 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 3.8/3.8 MB 14.2 MB/s eta 0:00:00
Using cached Deprecated-1.2.14-py2.py3-none-any.whl (9.6 kB)
Using cached iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Using cached packaging-23.2-py3-none-any.whl (53 kB)
Downloading cffi-1.16.0-cp39-cp39-manylinux_2_17_x86_64.manylinux2014_x86_64.whl (443 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 443.4/443.4 kB 15.8 MB/s eta 0:00:00
Downloading distlib-0.3.8-py2.py3-none-any.whl (468 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 468.9/468.9 kB 15.1 MB/s eta 0:00:00
Using cached filelock-3.13.1-py3-none-any.whl (11 kB)
Downloading importlib_metadata-7.0.2-py3-none-any.whl (24 kB)
Downloading platformdirs-4.2.0-py3-none-any.whl (17 kB)
Downloading wrapt-1.16.0-cp39-cp39-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_17_x86_64.manylinux2014_x86_64.whl (80 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 80.1/80.1 kB 16.5 MB/s eta 0:00:00
Using cached zipp-3.17.0-py3-none-any.whl (7.4 kB)
Building wheels for collected packages: detection_rules
  Building wheel for detection_rules (pyproject.toml) ... done
  Created wheel for detection_rules: filename=detection_rules-0.1.0-py3-none-any.whl size=33716432 sha256=a44f8c91a1fd698c96d77ce4f66b51b1f94c6ef1180ecac882763b114c0da726
  Stored in directory: /home/forteea1/.cache/pip/wheels/51/84/c6/d028d8b15b715615775cc1d1e482818ef141da32d3259c0354
Successfully built detection_rules
Installing collected packages: XlsxWriter, toml, pytoml, mccabe, jsl, distlib, zipp, wrapt, typing-extensions, tomli, semver, rpds-py, PyYAML, pyjwt, pyflakes, pycparser, pycodestyle, pluggy, platformdirs, packaging, nodeenv, mypy-extensions, marshmallow, marko, iniconfig, identify, filelock, exceptiongroup, Click, cfgv, attrs, virtualenv, typing-inspect, referencing, pytest, marshmallow-union, marshmallow-jsonschema, importlib-metadata, flake8, deprecated, cffi, typeguard, pynacl, pre-commit, marshmallow-dataclass, jsonschema-specifications, flake8-polyfill, PyGithub, pep8-naming, jsonschema, detection_rules
Successfully installed Click-8.1.7 PyGithub-1.55 PyYAML-6.0.1 XlsxWriter-1.3.9 attrs-23.2.0 cffi-1.16.0 cfgv-3.4.0 deprecated-1.2.14 detection_rules-0.1.0 distlib-0.3.8 exceptiongroup-1.2.0 filelock-3.13.1 flake8-3.8.1 flake8-polyfill-1.0.2 identify-2.5.35 importlib-metadata-7.0.2 iniconfig-2.0.0 jsl-0.2.4 jsonschema-4.21.1 jsonschema-specifications-2023.12.1 marko-2.0.1 marshmallow-3.13.0 marshmallow-dataclass-8.5.14 marshmallow-jsonschema-0.12.0 marshmallow-union-0.1.15.post1 mccabe-0.6.1 mypy-extensions-1.0.0 nodeenv-1.8.0 packaging-23.2 pep8-naming-0.7.0 platformdirs-4.2.0 pluggy-1.4.0 pre-commit-2.20.0 pycodestyle-2.6.0 pycparser-2.21 pyflakes-2.2.0 pyjwt-2.8.0 pynacl-1.5.0 pytest-8.0.2 pytoml-0.1.21 referencing-0.33.0 rpds-py-0.18.0 semver-3.0.0.dev4 toml-0.10.0 tomli-2.0.1 typeguard-3.0.2 typing-extensions-4.5.0 typing-inspect-0.8.0 virtualenv-20.25.1 wrapt-1.16.0 zipp-3.17.0

[notice] A new release of pip is available: 23.3.1 -> 24.0
[notice] To update, run: /home/forteea1/Code/clean_mains/detection-rules/env/detection-rules-build/bin/python -m pip install --upgrade pip
RELEASE: 
./env/detection-rules-build/bin/python -m detection_rules dev build-release --generate-navigator

█▀▀▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄▄▄ ▄   ▄      █▀▀▄ ▄  ▄ ▄   ▄▄▄ ▄▄▄
█  █ █▄▄  █  █▄▄ █    █   █  █ █ █▀▄ █      █▄▄▀ █  █ █   █▄▄ █▄▄
█▄▄▀ █▄▄  █  █▄▄ █▄▄  █  ▄█▄ █▄█ █ ▀▄█      █ ▀▄ █▄▄█ █▄▄ █▄▄ ▄▄█

[+] Building package 8.14
 - 5 rules excluded from package
Rule changes detected!
 - 71 changed rules
 - 0 new rules
 - 1 newly deprecated rules
run `build-release --update-version-lock` to update version.lock.json and deprecated_rules.json
Package saved to: /home/forteea1/Code/clean_mains/detection-rules/releases/8.14
loaded security_detection_engine manifests from the following package versions: ['8.13.1', '8.12.6', '8.12.5', '8.12.4', '8.12.3', '8.12.2', '8.12.1', '8.11.10', '8.11.9', '8.11.8', '8.11.7', '8.11.6', '8.11.5', '8.11.4', '8.11.3', '8.11.2', '8.11.1', '8.10.13', '8.10.12', '8.10.11', '8.10.10', '8.10.9', '8.10.8', '8.10.7', '8.10.6', '8.10.5', '8.10.4', '8.10.3', '8.10.2', '8.10.1', '8.9.15', '8.9.14', '8.9.13', '8.9.12', '8.9.11', '8.9.10', '8.9.9', '8.9.8', '8.9.7', '8.9.6', '8.9.5', '8.9.4', '8.9.3', '8.9.2', '8.9.1', '8.8.15', '8.8.14', '8.8.13', '8.8.12', '8.8.11', '8.8.10', '8.8.9', '8.8.8', '8.8.7', '8.8.6', '8.8.5', '8.8.4', '8.8.3', '8.8.2', '8.8.1', '8.7.13', '8.7.12', '8.7.11', '8.7.10', '8.7.9', '8.7.8', '8.7.7', '8.7.6', '8.7.5', '8.7.4', '8.7.3', '8.7.2', '8.7.1', '8.6.10', '8.6.9', '8.6.8', '8.6.7', '8.6.6', '8.6.5', '8.6.4', '8.6.3', '8.6.2', '8.6.1', '8.5.8', '8.5.7', '8.5.6', '8.5.5', '8.5.4', '8.5.3', '8.5.2', '8.5.1', '8.4.5', '8.4.4', '8.4.3', '8.4.2', '8.4.1', '8.3.4', '8.3.3', '8.3.2', '8.3.1', '8.2.1', '8.1.1', '1.0.2', '1.0.1']
[+] Adding historical rules from 8.13.1 package
- sha256: ae59d21b27e4f009bfc5ebbe844a2877fa6b435f64d096193c52059305ffd799
- 1075 rules included
rm -rf dist
mkdir dist
cp -r releases/*/*.zip dist/

detection-rules on  package_kql_kibana [!] is  v0.1.0 via  v3.8.18 (eql-rules-build) on  eric.forte took 4m11s 
❯ source env/detection-rules-build/bin/activate.fish

detection-rules on  package_kql_kibana [!] is  v0.1.0 via  v3.9.18 (detection-rules-build) on  eric.forte 
❯ pip install .[dev]
Looking in indexes: https://pypi.org/simple, https://eric.forte%40elastic.co:****@artifactory.elastic.dev/artifactory/api/pypi/pypi-endgame/simple
Processing /home/forteea1/Code/clean_mains/detection-rules
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Requirement already satisfied: Click~=8.1.0 in ./env/detection-rules-build/lib/python3.9/site-packages (8.1.7)
Requirement already satisfied: elasticsearch~=8.1 in ./env/detection-rules-build/lib/python3.9/site-packages (8.12.1)
Requirement already satisfied: eql==0.9.19 in ./env/detection-rules-build/lib/python3.9/site-packages (0.9.19)
Requirement already satisfied: jsl==0.2.4 in ./env/detection-rules-build/lib/python3.9/site-packages (0.2.4)
Requirement already satisfied: jsonschema>=3.2.0 in ./env/detection-rules-build/lib/python3.9/site-packages (4.21.1)
Requirement already satisfied: marko==2.0.1 in ./env/detection-rules-build/lib/python3.9/site-packages (2.0.1)
Requirement already satisfied: marshmallow-dataclass~=8.5.12 in ./env/detection-rules-build/lib/python3.9/site-packages (from marshmallow-dataclass[union]~=8.5.12) (8.5.14)
Requirement already satisfied: marshmallow-jsonschema~=0.12.0 in ./env/detection-rules-build/lib/python3.9/site-packages (0.12.0)
Requirement already satisfied: marshmallow-union~=0.1.15 in ./env/detection-rules-build/lib/python3.9/site-packages (0.1.15.post1)
Requirement already satisfied: marshmallow~=3.13.0 in ./env/detection-rules-build/lib/python3.9/site-packages (3.13.0)
Requirement already satisfied: pytoml==0.1.21 in ./env/detection-rules-build/lib/python3.9/site-packages (0.1.21)
Requirement already satisfied: requests~=2.27 in ./env/detection-rules-build/lib/python3.9/site-packages (2.31.0)
Requirement already satisfied: toml==0.10.0 in ./env/detection-rules-build/lib/python3.9/site-packages (0.10.0)
Requirement already satisfied: typing-inspect==0.8.0 in ./env/detection-rules-build/lib/python3.9/site-packages (0.8.0)
Requirement already satisfied: XlsxWriter~=1.3.6 in ./env/detection-rules-build/lib/python3.9/site-packages (1.3.9)
Requirement already satisfied: semver==3.0.0-dev.4 in ./env/detection-rules-build/lib/python3.9/site-packages (3.0.0.dev4)
Requirement already satisfied: typing-extensions==4.5.0 in ./env/detection-rules-build/lib/python3.9/site-packages (4.5.0)
Requirement already satisfied: PyYAML~=6.0.1 in ./env/detection-rules-build/lib/python3.9/site-packages (6.0.1)
Requirement already satisfied: pep8-naming==0.7.0 in ./env/detection-rules-build/lib/python3.9/site-packages (0.7.0)
Requirement already satisfied: PyGithub==1.55 in ./env/detection-rules-build/lib/python3.9/site-packages (1.55)
Requirement already satisfied: flake8==3.8.1 in ./env/detection-rules-build/lib/python3.9/site-packages (3.8.1)
Requirement already satisfied: pyflakes==2.2.0 in ./env/detection-rules-build/lib/python3.9/site-packages (2.2.0)
Requirement already satisfied: pytest>=3.6 in ./env/detection-rules-build/lib/python3.9/site-packages (8.0.2)
Requirement already satisfied: pre-commit==2.20.0 in ./env/detection-rules-build/lib/python3.9/site-packages (2.20.0)
Requirement already satisfied: lark-parser~=0.12.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from eql==0.9.19) (0.12.0)
Requirement already satisfied: pycodestyle<2.7.0,>=2.6.0a1 in ./env/detection-rules-build/lib/python3.9/site-packages (from flake8==3.8.1) (2.6.0)
Requirement already satisfied: mccabe<0.7.0,>=0.6.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from flake8==3.8.1) (0.6.1)
Requirement already satisfied: flake8-polyfill<2,>=1.0.2 in ./env/detection-rules-build/lib/python3.9/site-packages (from pep8-naming==0.7.0) (1.0.2)
Requirement already satisfied: cfgv>=2.0.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from pre-commit==2.20.0) (3.4.0)
Requirement already satisfied: identify>=1.0.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from pre-commit==2.20.0) (2.5.35)
Requirement already satisfied: nodeenv>=0.11.1 in ./env/detection-rules-build/lib/python3.9/site-packages (from pre-commit==2.20.0) (1.8.0)
Requirement already satisfied: virtualenv>=20.0.8 in ./env/detection-rules-build/lib/python3.9/site-packages (from pre-commit==2.20.0) (20.25.1)
Requirement already satisfied: deprecated in ./env/detection-rules-build/lib/python3.9/site-packages (from PyGithub==1.55) (1.2.14)
Requirement already satisfied: pyjwt>=2.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from PyGithub==1.55) (2.8.0)
Requirement already satisfied: pynacl>=1.4.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from PyGithub==1.55) (1.5.0)
Requirement already satisfied: mypy-extensions>=0.3.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from typing-inspect==0.8.0) (1.0.0)
Requirement already satisfied: elastic-transport<9,>=8 in ./env/detection-rules-build/lib/python3.9/site-packages (from elasticsearch~=8.1) (8.12.0)
Requirement already satisfied: attrs>=22.2.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from jsonschema>=3.2.0) (23.2.0)
Requirement already satisfied: jsonschema-specifications>=2023.03.6 in ./env/detection-rules-build/lib/python3.9/site-packages (from jsonschema>=3.2.0) (2023.12.1)
Requirement already satisfied: referencing>=0.28.4 in ./env/detection-rules-build/lib/python3.9/site-packages (from jsonschema>=3.2.0) (0.33.0)
Requirement already satisfied: rpds-py>=0.7.1 in ./env/detection-rules-build/lib/python3.9/site-packages (from jsonschema>=3.2.0) (0.18.0)
Requirement already satisfied: typeguard<4.0.0,>=2.4.1 in ./env/detection-rules-build/lib/python3.9/site-packages (from marshmallow-dataclass[union]~=8.5.12) (3.0.2)
Requirement already satisfied: iniconfig in ./env/detection-rules-build/lib/python3.9/site-packages (from pytest>=3.6) (2.0.0)
Requirement already satisfied: packaging in ./env/detection-rules-build/lib/python3.9/site-packages (from pytest>=3.6) (23.2)
Requirement already satisfied: pluggy<2.0,>=1.3.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from pytest>=3.6) (1.4.0)
Requirement already satisfied: exceptiongroup>=1.0.0rc8 in ./env/detection-rules-build/lib/python3.9/site-packages (from pytest>=3.6) (1.2.0)
Requirement already satisfied: tomli>=1.0.0 in ./env/detection-rules-build/lib/python3.9/site-packages (from pytest>=3.6) (2.0.1)
Requirement already satisfied: charset-normalizer<4,>=2 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (3.3.2)
Requirement already satisfied: idna<4,>=2.5 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (3.6)
Requirement already satisfied: urllib3<3,>=1.21.1 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (2.2.1)
Requirement already satisfied: certifi>=2017.4.17 in ./env/detection-rules-build/lib/python3.9/site-packages (from requests~=2.27) (2024.2.2)
Requirement already satisfied: setuptools in ./env/detection-rules-build/lib/python3.9/site-packages (from nodeenv>=0.11.1->pre-commit==2.20.0) (69.1.1)
Requirement already satisfied: cffi>=1.4.1 in ./env/detection-rules-build/lib/python3.9/site-packages (from pynacl>=1.4.0->PyGithub==1.55) (1.16.0)
Requirement already satisfied: importlib-metadata>=3.6 in ./env/detection-rules-build/lib/python3.9/site-packages (from typeguard<4.0.0,>=2.4.1->marshmallow-dataclass[union]~=8.5.12) (7.0.2)
Requirement already satisfied: distlib<1,>=0.3.7 in ./env/detection-rules-build/lib/python3.9/site-packages (from virtualenv>=20.0.8->pre-commit==2.20.0) (0.3.8)
Requirement already satisfied: filelock<4,>=3.12.2 in ./env/detection-rules-build/lib/python3.9/site-packages (from virtualenv>=20.0.8->pre-commit==2.20.0) (3.13.1)
Requirement already satisfied: platformdirs<5,>=3.9.1 in ./env/detection-rules-build/lib/python3.9/site-packages (from virtualenv>=20.0.8->pre-commit==2.20.0) (4.2.0)
Requirement already satisfied: wrapt<2,>=1.10 in ./env/detection-rules-build/lib/python3.9/site-packages (from deprecated->PyGithub==1.55) (1.16.0)
Requirement already satisfied: pycparser in ./env/detection-rules-build/lib/python3.9/site-packages (from cffi>=1.4.1->pynacl>=1.4.0->PyGithub==1.55) (2.21)
Requirement already satisfied: zipp>=0.5 in ./env/detection-rules-build/lib/python3.9/site-packages (from importlib-metadata>=3.6->typeguard<4.0.0,>=2.4.1->marshmallow-dataclass[union]~=8.5.12) (3.17.0)
Building wheels for collected packages: detection_rules
  Building wheel for detection_rules (pyproject.toml) ... done
  Created wheel for detection_rules: filename=detection_rules-0.1.0-py3-none-any.whl size=33716432 sha256=8fc432d0a801f177c3fe1fc8e33b2c73fd2bb01305bc0b2d35c8ffccb1f40700
  Stored in directory: /home/forteea1/.cache/pip/wheels/51/84/c6/d028d8b15b715615775cc1d1e482818ef141da32d3259c0354
Successfully built detection_rules
Installing collected packages: detection_rules
  Attempting uninstall: detection_rules
    Found existing installation: detection_rules 0.1.0
    Uninstalling detection_rules-0.1.0:
      Successfully uninstalled detection_rules-0.1.0
Successfully installed detection_rules-0.1.0

[notice] A new release of pip is available: 23.3.1 -> 24.0
[notice] To update, run: pip install --upgrade pip

detection-rules on  package_kql_kibana [!] is  v0.1.0 via  v3.9.18 (detection-rules-build) on  eric.forte took 7s 

```

</p>
</details>

Reply:
I actually meant to link the PyYAML change a few lines above.